### PR TITLE
Recipe refactor

### PR DESCRIPTION
**Pull request by DaddyWesker**
I've added function to set only recipes for items we want. Though there is a problem i wanted to discuss. Currently I'm assuming that person who wants to add a recipe for some item can send something like "pickaxe" and code will add every pickaxe (or, in current implementation, every item with _pickaxe at the end), which could be undesirable. Or will it better to use strict names and strict match?

### CODE REVIEW

**Comment by Necr0x0Der** [comment]
I'd suppose strict name here. You can add a flag to the function to indicate that name suffixes are provided instead of strict names, but strict names should be used by default.

**Comment by DaddyWesker** [comment]
@Necr0x0Der  Done

**Comment by Necr0x0Der** [comment]
Is it possible with replacing `self.crafts = ...` in `minelogy` with this approach?

File: examples/minelogy.py [hunk 1]
```diff
@@ -181,31 +181,48 @@ def set_recipes(self, mcrecipes):
...
+        if clear_recipes:
+            self.crafts = []
+        for mcrecipe in mcrecipes:
+            craft_name = mcrecipe['name'].split('.')[-1]
+            if (craft_name in items_to_add) or (not strict_matching and craft_name.split("_")[-1] in items_to_add):
+                craft_entity = self.set_one_recipe(mcrecipe, craft_name)
+                if craft_entity is not None:
+                    self.crafts.append(craft_entity)
+
+    def set_one_recipe(self, mcrecipe, craft_name):
```
Comment (noskill) [inline examples/minelogy.py:198]: Could you please add typing and/or a docstring? It's not clear where the recipe is to be set..

File: tagilmo/utils/vereya_wrapper.py [hunk 2]
```diff
@@ -252,13 +252,18 @@ def getChat(self, nAgent=0):
         else:
             return None
 
-    def getItemList(self, nAgent=0):
+    def getItemAndRecipeList(self, nAgent=0):
         self.sendCommand('item_list')
+        self.sendCommand('recipes')
         time.sleep(2)
```
Comment (noskill) [inline tagilmo/utils/vereya_wrapper.py:258]: sleeps are not robust, you can build api on waitNotNoneObserve instead

Comment (DaddyWesker) [reply tagilmo/utils/vereya_wrapper.py:258]: weird, but waitNotNoneObserver is currently returning None when i'm trying to implement getItemAndRecipeList using it.

File: tests/vereya/test_observation.py [hunk 3]
```diff
@@ -76,8 +76,9 @@ def test_observation_from_chat(self):
         self.assertEqual(command[0], "get wooden_axe")
 
     def test_observation_from_item(self):
-        item_list = self.rob.mc.getItemList()
+        item_list, recipes = self.rob.mc.getItemAndRecipeList()
```
Comment (noskill) [inline tests/vereya/test_observation.py:79]: does it work without calling observeProc?

Comment (DaddyWesker) [reply tests/vereya/test_observation.py]: @noskill  fixed. Now works

**Review by noskill — APPROVED** [review]

**Review by Necr0x0Der — APPROVED** [review]

**Review by CICS-Oleg — APPROVED** [review]

**Pull request merged**